tests.sh

```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
+
+# Default script used by the Docker testing harness in the container run by ./docker-tests.sh
+
+# Strict, and loud failure
+set -euo pipefail
+trap 'rc=$?;set +ex;if [[ $rc -ne 0 ]];then trap - ERR EXIT;echo 1>&2;echo "*** fail *** : code $rc : $DIR/$SCRIPT $ARGS" 1>&2;echo 1>&2;exit $rc;fi' ERR EXIT
+ARGS="$*"
+DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+SCRIPT="$(basename "${BASH_SOURCE[0]}")"
+
+set -x
+
+# Dev-centric test steps based on scripts in package.json
+PATH=$(npm bin):$PATH
+
+# If ava testing hangs (which, sadly, happens all too often), add --serial to get a more
+# deterministic record that will log up to the test prior to the one that hangs... But, as
+# is often the case with race conditions, the instrumentation may eliminate the problem
+# you're trying to track down...
+ava --no-color --timeout 30000 --verbose
+eslint .
+prettier --check '**/*.(html|json|md|sublime-project|ts|yml)'
+nyc ava
```

Comment:
This will re-run ava - what do you think about combining this with the initial `ava` invocation?

Reply:
We should discuss the purpose of this or other scripts we include.  As it stands, this script supports a dev-cycle (my dev cycle) whereby you get stuff working, you mollify the styling police, you move towards deployable coverage, etc.  I've gone back and forth on just doing `npm test`, but it fails too often and the output isn't that helpful when you start on forensic work.

Reply:
I'm specifically wondering about the inclusion of both `ava` and `nyc ava` in this script - not the use of a separate `tests.sh` file. Not sure if that was apparent from my original comment.